plugins/powerd/fu-plugin-powerd.c

```diff
@@ -20,6 +20,41 @@ fu_plugin_init(FuPlugin *plugin)
 	fu_plugin_alloc_data(plugin, sizeof(FuPluginData));
 }
 
+static gboolean
+fu_plugin_powerd_create_suspend_file(GError **error)
+{
+	g_autofree gchar *lockdir = NULL;
+	g_autofree gchar *inhibitsuspend_filename = NULL;
+	g_autofree gchar *getpid_str = NULL;
+	gboolean file_present = FALSE;
+
+	lockdir = fu_common_get_path(FU_PATH_KIND_LOCKDIR);
+	inhibitsuspend_filename = g_build_filename(lockdir, "power_override", "fwupd.lock", NULL);
+	getpid_str = g_strdup_printf("%d", getpid());
+	file_present = g_file_set_contents(inhibitsuspend_filename, getpid_str, -1, error);
+	if (!file_present)
+		g_prefix_error(error, "lock file unable to be created");
+	return file_present;
+}
+
+static gboolean
+fu_plugin_powerd_delete_suspend_file(GError **error)
+{
+	g_autofree gchar *lockdir = NULL;
+	g_autoptr(GFile) inhibitsuspend_file = NULL;
+	gboolean file_absent = TRUE;
+
+	lockdir = fu_common_get_path(FU_PATH_KIND_LOCKDIR);
+	inhibitsuspend_file =
+	    g_file_new_build_filename(lockdir, "power_override", "fwupd.lock", NULL);
+	if (g_file_query_exists(inhibitsuspend_file, NULL)) {
+		file_absent = g_file_delete(inhibitsuspend_file, NULL, error);
```

Comment:
What about (here and above): 
```
...{
if (!g_file_delete(inhibitsuspend_file, NULL, error)) {
    g_prefix_error(error, "lock file unable to be deleted");
    return FALSE;
}
}
return TRUE;
```

Reply:
If g_file_delete () can't delete, it returns false, but in the case that the file doesn't exist, it also raises an error. I added a check so that, if it returns false and the error is due to something other than the file not existing already, it will return false.

Documentation did suggest that using the query was not as good as just deleting and dealing with the error if it arises (in case something happens to the file between the check and the command).